apps/crn-server/src/data-providers/contentful/team.data-provider.ts

```diff
@@ -172,7 +186,59 @@ export class TeamContentfulDataProvider implements TeamDataProvider {
   }
 }
 
-export const parseContentfulGraphQlTeams = (item: TeamItem): TeamDataObject => {
+export const parseContentfulGraphQlTeams = (
+  item: TeamItem,
+): TeamListItemDataObject => {
+  const expertiseAndResourceTags = (item.expertiseAndResourceTags || []).reduce(
+    (tags: string[], tag) => {
+      if (tag) {
+        tags.push(tag);
+      }
+      return tags;
+    },
+    [],
+  );
+
+  const [numberOfMembers, labIds]: [number, Set<string>] = (
+    item.linkedFrom?.teamMembershipCollection?.items || []
+  ).reduce(
+    ([memberCount, labIdsSet], membership: Membership | null) => {
+      if (
+        !membership ||
+        !membership.linkedFrom?.usersCollection?.items[0]?.onboarded ||
+        !membership.role
+      ) {
+        return [memberCount, labIdsSet];
+      }
+
+      const { labsCollection } =
+        membership.linkedFrom?.usersCollection?.items[0] || {};
+
+      const memberLabIds = labsCollection?.items
+        .map((labItem) => labItem?.sys.id)
+        .filter((x): x is string => x !== undefined);
+
+      memberLabIds?.forEach((labId) => labIdsSet.add(labId));
+
+      return [memberCount + 1, labIdsSet];
+    },
+    [0, new Set() as Set<string>],
+  );
```

Comment:
this seems a little complicated, in a sense that something similar could probably be achieved with some flatMap and length, even if you were to do it twice (the optimisation seems unnecessary anyway, even if we are processing this for each team). Leave it but in the future I would start off simple and then optimise like this when really needed.